egs/callhome_diarization/v1/local/dump_model.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import numpy as np
+import pickle
+import sys
+
+def load_dubm(dubm_text):
+    para_dict = {}
+    with open(dubm_text, 'r') as fh:
+        content = fh.readlines()
+    state = 0
+    data_array = []
+
+    for line in content:
+        line = line.strip('\n')
+        line_split = line.split()
+        if state == 0:
+            if len(line_split) == 1:
+                continue
+            elif len(line_split) == 2 and line_split[1] == "[":
+                para_name = line_split[0]
+                state = 1
+                data_array = []
+            elif len(line_split) >= 3 and line_split[1] == "[" and line_split[-1] == "]": # One line vector
+                para_name = line_split[0]
+                data_list = []
+                for i in range(2, len(line_split) - 1):
+                    data_list.append(float(line_split[i]))
+                data_list = np.array(data_list)
+                para_dict[para_name] = data_list
+            else:
+                raise ValueError("Condition not defined.")
+        elif state == 1:
+            if line_split[-1] == "]":
+                data_list = []
+                for i in range(len(line_split) - 1):
+                    data_list.append(float(line_split[i]))
+                data_list = np.array(data_list)
+                data_array.append(data_list)
+                data_array = np.array(data_array)
+                para_dict[para_name] = data_array
+                state = 0
+            else:
+                data_list = []
+                for i in range(len(line_split)):
+                    data_list.append(float(line_split[i]))
+                data_list = np.array(data_list)
+                data_array.append(data_list)
+        else:
+            raise ValueError("Condition not defined.")
+    return para_dict 
+
+def load_ivector_extractor(ie_text):
+    para_dict = {}
+    with open(ie_text, 'r') as fh:
+        content = fh.readlines()
+    state = 0
+    data_3dmatrix = []
+    data_matrix = []
+    data_array = []
+
+    for line in content:
+        line = line.strip('\n')
+        if line == "<SigmaInv> [":
+            break
+        if state == 0:
+            if line != "<M> 1024  [":
```

Comment:
What is the 1024 referencing? Should this be configurable?

Reply:
I am not sure about this.

Reply:
@saikiranvalluri, where'd this code come from originally?

Reply:
Nagendra sir provided me the scripts dump_model.py and kaldi_io.py